app/forms/csv_upload_form.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CSVUploadForm
+  include ActiveModel::Validations
+
+  attr_reader :file
+
+  validates_with CSVValidator
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(params)
+    @file = params[:csv_upload_form][:file]
+
+    return false unless valid?
+
+    begin
+      Keyword.transaction do
+        # rubocop:disable Rails/SkipsModelValidations
+        user.keywords.insert_all parsed_keywords
+        # rubocop:enable Rails/SkipsModelValidations
+      end
+    rescue ActiveRecord::StatementInvalid
+      errors.add(:keyword, I18n.t('csv.validation.bad_keyword_length'))
+    end
+
+    errors.empty?
+  end
+
+  private
+
+  attr_reader :user
+
+  def parsed_keywords
+    time = Time.current
+    CSV.read(file.tempfile).filter_map do |row|
+      keyword_attributes row.join(','), time
+    end
+  end
+
+  def keyword_attributes(name, time)
+    return nil if name.blank?
+
+    {
+      user_id: user.id,
+      name: name,
+      created_at: time,
+      updated_at: time
```

Comment:
Rails handle for us setting `created_at`, `updated_at `
```suggestion
```

Reply:
@junan : Not when using `insert_all`, it would raise an error~ 
I had the same though but trials and fail proved me the opposite ^^

Reply:
Though [this discussion](https://github.com/rails/rails/issues/35493) points to an interesting workarround: I can add an explicit default value to the DB columns.
I think it's an interesting move... @junan & @ankitkalia1195 do you see any issue with that solution? :)

Reply:
@malparty Yea the solution seems good

Reply:
@ankitkalia1195 @junan 
Solution implemented [here](https://github.com/malparty/google-search-ruby/pull/52/commits/6e5d5a5b103dd69b395fbf5f9d80f1813e7799bb). 
For migration, I could not find a ruby-way to handle it properly, so I used SQL execute. I hope it's okay. 

Thanks :)